### PR TITLE
Update dependencies, add auto artifact upload

### PR DESCRIPTION
**Pull request by Coccocoahelper**
FINALLY!!!!

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,19 +10,23 @@ jobs:
...
-
       - name: Grant execute permission
         run: chmod +x ./gradlew
       - name: Build with Gradle
         run: ./gradlew build
+      - name: Upload a Build Artifact
+        uses: actions/upload-artifact@v4.3.0
+        with: 
+          name: LegacyFixes
+          path: build/libs/*.jar
```
Comment (BluCobalt) [inline .github/workflows/build.yml:32]: Why call it legacy fixes?

**Review by BluCobalt — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Coccocoahelper) [reply .github/workflows/build.yml:32]: My bad, fixing it ASAP.

Comment (Coccocoahelper) [reply .github/workflows/build.yml:32]: @BluCobalt Done, ready to merge.

**Review by thecatcore — APPROVED** [review]
lgtm

**Review by BluCobalt — APPROVED** [review]

**Pull request merged**